cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -144,6 +197,71 @@ protected void runOneIteration() throws Exception {
   }
 
   /**
+   * Perform aggregation on the Streams described by the {@code streamNames}, and no other Streams.
+   * If aggregation was previously done on other Streams, those must be cancelled.
+   *
+   * @param streamNames names of the streams to perform data sizes aggregation on
+   */
+  public void aggregate(Set<String> streamNames) {
+    Set<String> existingAggregators = Sets.newHashSet(aggregators.keySet());
+    for (String streamName : streamNames) {
+      if (existingAggregators.remove(streamName)) {
+        continue;
+      }
+
+      long filesSize = 0;
+      try {
+        StreamConfig config = streamAdmin.getConfig(streamName);
+        filesSize = StreamUtils.fetchStreamFilesSize(config);
+      } catch (IOException e) {
+        LOG.error("Could not compute sizes of files for stream {}", streamName);
+      }
+
+      Aggregator aggregator = new Aggregator(streamName, filesSize);
+      aggregator.startAndWait();
```

Comment:
So there is one Aggregate per stream in the leader of that stream, and it periodically aggregates the heartbeats received for the stream, to publish a notification.